cmd/ctr/commands/images/pull.go

```diff
@@ -265,7 +266,7 @@ func ProgressHandler(ctx context.Context, out io.Writer) (transfer.ProgressFunc,
 		statuses = map[string]*progressNode{}
 		roots    = []*progressNode{}
 		progress transfer.ProgressFunc
-		pc       = make(chan transfer.Progress, 1)
+		pc       = make(chan transfer.Progress, 5)
```

Comment:
This number seems to need an explanation

Reply:
Added, although this is still kind of a bandaid solution. Right now the progress drain is a bit racy and we could probably add a better solution, especially if we could just rely on some indicator in the progress stream that the end has been reached (with a timeout accounting for it not being reached).